Trisha-tech/OnlineBookSales
# Add Dark Mode Feature

### Description
This PR adds a dark mode feature to the OnlineBookSales website to reduce eye strain during nighttime browsing.

- This PR does the following:
  - Adds a dark mode toggle switch in the website's header.
  - Updates CSS to support dark mode styling.
  - Ensures text remains visible in both light and dark modes by setting explicit text colors where necessary.

### Related Issues
Fixes #177

### Changes
- Added a new feature to toggle dark mode in `App.js`.
- Updated `Navbar` component to include a dark mode toggle button.
- Modified CSS to support dark mode and light mode styling.
- Ensured critical text remains visible by adding specific classes to keep text black in dark mode.

### Testing Instructions
1. Pull this branch.
2. Run `npm install` to install dependencies.
3. Run the application with `npm start`.
4. Verify that the dark mode toggle button appears in the header.
5. Click the toggle button to switch between dark mode and light mode.
6. Ensure that text visibility is maintained in both modes.

### Screenshots (if applicable)
![Light Mode Screenshot](https://github.com/Trisha-tech/OnlineBookSales/assets/109446288/19693181-6266-4218-b133-1d7310631d6c)

![Dark Mode Screenshot]
![Screenshot 2024-06-21 121215](https://github.com/Trisha-tech/OnlineBookSales/assets/109446288/99d1b7bd-ba83-4d5b-b699-4c215fc63a77)


### Additional Context
- This PR is based on the request raised in issue #177.
- Worked on this issue under GSSOC'24.

### Checklist
